ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -276,13 +295,34 @@ public QueryMetadata getQueryExecutionPlan(final Query query) throws Exception {
               querySpecification,
               createTableAsSelect.getName().getSuffix(),
               createTableAsSelect.getProperties(),
-              Optional.empty()
+              Optional.empty(),
+              true
       );
 
       tempMetaStoreForParser.putSource(queryEngine.getResultDatasource(
               querySpecification.getSelect(),
               createTableAsSelect.getName().getSuffix()
       ).cloneWithTimeKeyColumns());
+      return new Pair<>(statementString, query);
+    } else if (statement instanceof InsertInto) {
+      InsertInto insertInto = (InsertInto) statement;
+      if (tempMetaStoreForParser.getSource(insertInto.getTarget().getSuffix().toString()) == null) {
+        throw new KsqlException(String.format("Sink, %s, does not exist for the INSERT INTO "
+                                              + "statement.", insertInto.getTarget().getSuffix()
+            .toString()));
+      }
+      QuerySpecification querySpecification =
+          (QuerySpecification) insertInto.getQuery().getQueryBody();
+
+      Query query = addInto(
+          insertInto.getQuery(),
+          querySpecification,
+          insertInto.getTarget().getSuffix(),
+          new HashMap<>(),
+          insertInto.getPartitionByColumn(),
```

Comment:
I think this is incorrect, i.e, the partition by column should always be the same as that of the `create stream as ...`.